sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -0,0 +1,226 @@
+                # so once we hit the max bytes limit, return
+                return normalized_rows
+            normalized_rows.append(row)
+        return normalized_rows
+
+    @staticmethod
+    def _sanitize_row(row, obfuscated_statement):
+        row = {key: val for key, val in row.items() if key not in dm_exec_requests_exclude_keys and val is not None}
+        row['text'] = obfuscated_statement
+        if 'query_hash' in row:
+            row['query_hash'] = _hash_to_hex(row['query_hash'])
+        if 'query_plan_hash' in row:
+            row['query_plan_hash'] = _hash_to_hex(row['query_plan_hash'])
+        return row
+
+    @staticmethod
+    def _get_estimated_row_size_bytes(row):
+        return len(str(row))
+
+    def _create_activity_event(self, active_sessions, active_connections):
+        event = {
+            "host": self._db_hostname,
+            "ddagentversion": datadog_agent.get_version(),
+            "ddsource": "sqlserver",
+            "dbm_type": "activity",
+            "collection_interval": self.collection_interval,
+            "ddtags": self.check.tags,
+            "timestamp": time.time() * 1000,
+            "sqlserver_activity": active_sessions,
+            "sqlserver_connections": active_connections,
+        }
+        return event
+
+    def _truncate_activity_rows(self, rows, max_bytes):
+        pass
+
+    def collect_activity(self):
+        """
+        Collects all current activity for the SQLServer intance.
+        :return:
+        """
+        start_time = time.time()
+        try:
+            # re-use the check's conn module, but set extra_key=dbm-activity- to ensure we get our own
+            # raw connection. adodbapi and pyodbc modules are thread safe, but connections are not.
+            with self.check.connection.open_managed_default_connection(key_prefix=self._conn_key_prefix):
+                with self.check.connection.get_managed_cursor(key_prefix=self._conn_key_prefix) as cursor:
+                    connections = self._get_active_connections(cursor)
+                    rows = self._get_activity(cursor)
+                    normalized_rows = self._normalize_queries_and_filter_rows(rows, MAX_PAYLOAD_BYTES)
+                    event = self._create_activity_event(normalized_rows, connections)
+                    self._check.database_monitoring_query_activity(
+                        json.dumps(event, default=default_json_event_encoding)
+                    )
+        except Exception:
+            self.log.exception('Unable to collect activity due to an error')
+            self.check.count(
+                "dd.sqlserver.activity.error",
+                1,
+                tags=self.check.debug_tags(),
+                hostname=self.check.resolved_hostname,
+            )
+            return []
+
+        elapsed_ms = (time.time() - start_time) * 1000
+        self.check.histogram(
+            "dd.sqlserver.activity.collect_activity.time",
```

Comment:
We probably only want to track this time on successful runs? So if there's an unhandled exception then we don't submit this?

Reply:
@djova do you think its worth adding metrics around the size of the payloads? At least while this is in beta testing with customers?

Reply:
I added a histo metric for this in this commit dc44f98928791fd8a220f79951915d1582c701a5